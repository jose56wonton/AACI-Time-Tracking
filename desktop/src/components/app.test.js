import React from 'react';
import { shallow } from 'enzyme';

import App,{ AuthedRoutes } from 'app';

const props = {};

const setup = () => {  
  return shallow(<AuthedRoutes {...props} />);    
};

const setupHOC = () => {
  return shallow(<App {...props} />);
};

describe('App', () => {
  it('should render correctly', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();    
  });
  it('should render correctly withRouter', () => {
    const wrapper = setupHOC();
    expect(wrapper).toMatchSnapshot();
  });
});