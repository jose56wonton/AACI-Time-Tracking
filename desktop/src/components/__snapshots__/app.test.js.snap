// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Components should render correctly 1`] = `
<div>
  <Connect(SnackRootContainer) />
  <Connect(RootModalContainer) />
  <Component
    component={[Function]}
    key="0"
    name="Signin"
    path="/"
    routes={
      Array [
        Object {
          "component": [Function],
          "exact": true,
          "name": "Signin",
          "path": "/",
        },
      ]
    }
  />
  <Component
    component={[Function]}
    key="1"
    name="Employee"
    path="/employee"
    routes={
      Array [
        Object {
          "component": [Function],
          "exact": true,
          "name": "Employee Actions",
          "path": "/employee",
          "type": "employee",
        },
        Object {
          "component": [Function],
          "exact": true,
          "name": "Employee Clockout",
          "path": "/employee/clockout",
          "type": "employee",
        },
      ]
    }
  />
  <Component
    component={[Function]}
    key="2"
    name="Manager"
    path="/manager"
    routes={
      Array [
        Object {
          "component": [Function],
          "exact": true,
          "name": "Manager Actions",
          "path": "/manager",
          "type": "manager",
        },
        Object {
          "component": [Function],
          "exact": true,
          "name": "Manager Clockout",
          "path": "/manager/clockout",
          "type": "manager",
        },
        Object {
          "component": [Function],
          "exact": true,
          "name": "Manager Export",
          "path": "/manager/export",
          "type": "manager",
        },
      ]
    }
  />
  <Component
    component={[Function]}
    key="3"
    name="Admin"
    path="/admin"
    routes={
      Array [
        Object {
          "component": [Function],
          "exact": true,
          "name": "Admin Actions",
          "path": "/admin",
          "type": "admin",
        },
        Object {
          "component": [Function],
          "exact": true,
          "name": "Admin Clockout",
          "path": "/admin/clockout",
          "type": "admin",
        },
        Object {
          "component": [Function],
          "exact": true,
          "name": "Admin Export",
          "path": "/admin/export",
          "type": "admin",
        },
        Object {
          "component": [Function],
          "exact": true,
          "name": "Admin Analyze",
          "path": "/admin/analyze",
          "type": "admin",
        },
      ]
    }
  />
</div>
`;

exports[`App Components should render correctly withRouter 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      component={[Function]}
      path="/auth"
    />
    <Connect(PrivateRoute)
      component={[Function]}
      path="/"
    />
  </Switch>
</BrowserRouter>
`;

exports[`App should render correctly 1`] = `
<div>
  <Connect(SnackRootContainer) />
  <Connect(RootModalContainer) />
  <Component
    component={[Function]}
    key="0"
    name="Signin"
    path="/"
    routes={
      Array [
        Object {
          "component": [Function],
          "exact": true,
          "name": "Signin",
          "path": "/",
        },
      ]
    }
  />
  <Component
    component={[Function]}
    key="1"
    name="Employee"
    path="/employee"
    routes={
      Array [
        Object {
          "component": [Function],
          "exact": true,
          "name": "Employee Actions",
          "path": "/employee",
          "type": "employee",
        },
        Object {
          "component": [Function],
          "exact": true,
          "name": "Employee Clockout",
          "path": "/employee/clockout",
          "type": "employee",
        },
      ]
    }
  />
  <Component
    component={[Function]}
    key="2"
    name="Manager"
    path="/manager"
    routes={
      Array [
        Object {
          "component": [Function],
          "exact": true,
          "name": "Manager Actions",
          "path": "/manager",
          "type": "manager",
        },
        Object {
          "component": [Function],
          "exact": true,
          "name": "Manager Clockout",
          "path": "/manager/clockout",
          "type": "manager",
        },
        Object {
          "component": [Function],
          "exact": true,
          "name": "Manager Export",
          "path": "/manager/export",
          "type": "manager",
        },
      ]
    }
  />
  <Component
    component={[Function]}
    key="3"
    name="Admin"
    path="/admin"
    routes={
      Array [
        Object {
          "component": [Function],
          "exact": true,
          "name": "Admin Actions",
          "path": "/admin",
          "type": "admin",
        },
        Object {
          "component": [Function],
          "exact": true,
          "name": "Admin Clockout",
          "path": "/admin/clockout",
          "type": "admin",
        },
        Object {
          "component": [Function],
          "exact": true,
          "name": "Admin Export",
          "path": "/admin/export",
          "type": "admin",
        },
        Object {
          "component": [Function],
          "exact": true,
          "name": "Admin Analyze",
          "path": "/admin/analyze",
          "type": "admin",
        },
      ]
    }
  />
</div>
`;

exports[`App should render correctly withRouter 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      component={[Function]}
      path="/auth"
    />
    <Connect(PrivateRoute)
      component={[Function]}
      path="/"
    />
  </Switch>
</BrowserRouter>
`;
